### ISSUE TITLE
Cannot run app - constant crashes on app start

### ISSUE THREAD

**Issue by alexeyvasilyev** [martoreto/aauto-sdk-demo#2]
App compiled against com.github.martoreto:aauto-sdk:v4.2 causes constant crashes on start. Just black screen is shown.

```
11-28 20:02:00.442 4777-17284/? I/ActivityManager: Start proc 13974:com.github.martoreto.aademo/u0a670 for service com.github.martoreto.aademo/.CarService
11-28 20:02:00.592 13974-13974/com.github.martoreto.aademo I/ApiFactory: Initialized ApiFactory to load from remote APK
11-28 20:02:00.896 13974-13974/com.github.martoreto.aademo W/zygote64: Unsupported class loader
11-28 20:02:00.904 13974-13974/com.github.martoreto.aademo I/DynamicApiFactory: Will load car API impls dynamically from com.google.android.projection.gearhead
11-28 20:02:00.918 13974-13974/com.github.martoreto.aademo I/CAR.CLIENT: SdkVersion: 4 ImplVersion: 4
11-28 20:02:01.015 13974-13987/com.github.martoreto.aademo W/DynamiteModule: Local module descriptor class for com.google.android.gms.googlecertificates not found.
11-28 20:02:01.020 13974-13987/com.github.martoreto.aademo W/zygote64: Unsupported class loader
11-28 20:02:01.022 13974-13987/com.github.martoreto.aademo W/zygote64: Skipping duplicate class check due to unsupported classloader
11-28 20:02:01.024 13974-13987/com.github.martoreto.aademo I/DynamiteModule: Considering local module com.google.android.gms.googlecertificates:0 and remote module com.google.android.gms.googlecertificates:4
11-28 20:02:01.024 13974-13987/com.github.martoreto.aademo I/DynamiteModule: Selected remote version of com.google.android.gms.googlecertificates, version >= 4
11-28 20:02:01.047 13974-13987/com.github.martoreto.aademo W/zygote64: Unsupported class loader
11-28 20:02:01.048 13974-13987/com.github.martoreto.aademo W/zygote64: Skipping duplicate class check due to unsupported classloader
11-28 20:02:01.127 4777-17287/? I/DisplayManagerService: Display device added: DisplayDeviceInfo{"com.github.martoreto.aademo/com.github.martoreto.aademo.MainCarActivity": uniqueId="virtual:com.github.martoreto.aademo,10670,com.github.martoreto.aademo/com.github.martoreto.aademo.MainCarActivity,0", 800 x 400, modeId 41, defaultModeId 41, supportedModes [{id=41, width=800, height=400, fps=60.0}], colorMode 0, supportedColorModes [0], HdrCapabilities null, density 160, 160.0 x 160.0 dpi, appVsyncOff 0, presDeadline 16666666, touch NONE, rotation 0, type VIRTUAL, state ON, owner com.github.martoreto.aademo (uid 10670), FLAG_PRIVATE, FLAG_NEVER_BLANK, FLAG_PRESENTATION, FLAG_OWN_CONTENT_ONLY}
11-28 20:02:01.128 13974-13974/com.github.martoreto.aademo D/AndroidRuntime: Shutting down VM
11-28 20:02:01.131 13974-13974/com.github.martoreto.aademo E/AndroidRuntime: FATAL EXCEPTION: main
                                                                             Process: com.github.martoreto.aademo, PID: 13974
                                                                             java.lang.RuntimeException: Exception while instantiating class com.github.martoreto.aademo.MainCarActivity
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:429)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:205)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:595)
                                                                                 at com.google.android.gms.car.internal.i.run(CarActivityHostImpl.java:2)
                                                                                 at android.os.Handler.handleCallback(Handler.java:790)
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                 at android.os.Looper.loop(Looper.java:164)
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                                 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                              Caused by: java.lang.NoSuchMethodException: <init> []
                                                                                 at java.lang.Class.getConstructor0(Class.java:2320)
                                                                                 at java.lang.Class.getConstructor(Class.java:1725)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:426)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:205) 
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:595) 
                                                                                 at com.google.android.gms.car.internal.i.run(CarActivityHostImpl.java:2) 
                                                                                 at android.os.Handler.handleCallback(Handler.java:790) 
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                 at android.os.Looper.loop(Looper.java:164) 
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                                 at java.lang.reflect.Method.invoke(Native Method) 
                                                                                 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
11-28 20:02:01.133 13974-13974/com.github.martoreto.aademo E/CAR.CLIENT.EXN: FATAL EXCEPTION: main
                                                                             Process: com.github.martoreto.aademo
                                                                             PID: 13974
                                                                             
                                                                             java.lang.RuntimeException: Exception while instantiating class com.github.martoreto.aademo.MainCarActivity
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:429)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:205)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:595)
                                                                                 at com.google.android.gms.car.internal.i.run(CarActivityHostImpl.java:2)
                                                                                 at android.os.Handler.handleCallback(Handler.java:790)
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                 at android.os.Looper.loop(Looper.java:164)
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                                 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                              Caused by: java.lang.NoSuchMethodException: <init> []
                                                                                 at java.lang.Class.getConstructor0(Class.java:2320)
                                                                                 at java.lang.Class.getConstructor(Class.java:1725)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:426)
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:205) 
                                                                                 at com.google.android.gms.car.internal.CarActivityHostImpl.a(CarActivityHostImpl.java:595) 
                                                                                 at com.google.android.gms.car.internal.i.run(CarActivityHostImpl.java:2) 
                                                                                 at android.os.Handler.handleCallback(Handler.java:790) 
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                 at android.os.Looper.loop(Looper.java:164) 
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                                 at java.lang.reflect.Method.invoke(Native Method) 
                                                                                 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
```

**Comment by jilleb** [comment]
Same here.

**Comment by martoreto** [comment]
This happens when MainCarActivity is not a public class. This is fixed in 2fe91a701ddde7578d19a2632cf1f0edd288b226.
